test/models/user_test.rb

```diff
@@ -0,0 +1,53 @@
+require "test_helper"
+
+describe User do
+  describe "validations" do
+    before do
+      @user = User.new(username: "bananapants")
+    end
+
+    it "creates a user when given valid input" do
+      success = @user.valid?
+
+      expect(success).must_equal true
+    end
+
+    it "won't create a user without a username" do
+      @user.username = nil
+
+      success = @user.valid?
+
+      expect(success).must_equal false
+    end
+
+    it "won't create a user with a repeat name" do
+      @user.save
+      clone_user = User.new(username: "bananapants")
+
+      success = clone_user.valid?
+
+      expect(success).must_equal false
+    end
+  end
+
+  describe "relationships" do
+    before do
+      @work = works(:book12)
+      @user = users(:testuser)
+      @vote = Vote.new(user: @user, work: @work)
+    end
+
+    it "can have a vote" do
+      @vote.save
```

Comment:
I recommend doing `save!` or asserting on the response so to make it easier to debug test failures.